spi/core-spi/src/test/java/org/eclipse/dataspaceconnector/spi/types/domain/DataAddressTest.java

```diff
@@ -60,4 +58,48 @@ void verifyNullKeyThrowsException() {
                 .hasMessageContaining("Property key null.");
     }
 
+    @Test
+    void verifyCopy() {
+        DataAddress dataAddress = newSampleDataAddress();
+
+        var copy = dataAddress.copy();
+
+        assertSampleDataAddress(dataAddress, copy);
+    }
+
+    @Test
+    void verifyDeepCopy() {
+        DataAddress dataAddress = newSampleDataAddress();
+
+        var copy = dataAddress.copy();
+
+        var copyProperties = copy.getProperties();
+        copyProperties.put("foo", "new value");
+
+        assertEquals("bar", dataAddress.getProperty("foo"));
```

Comment:
mixing AssertJ and Jupiter assertions makes the code harder to read. I would recommend to use only AssertJ.

```
assertEquals -> assertThat(...).isEqualTo(...)
assertNotNull ->  assertThat(...).isNotNull()
assert